nltools/data.py

```diff
@@ -1410,13 +1410,13 @@ def aggregate(self, mask, func):
         values = dat.apply(func)
         return dat.combine(values)
 
-    def threshold(self, thresh=0, binarize=False):
-        '''Threshold Brain_Data instance
+    def threshold(self, thresh=0, binarize=False, upper_only=False):
+        '''Threshold Brain_Data instance. If data contain both positive and negative values, defaults to two-sided thresholding automatically. This can be changed with the upper_only flag.
```

Comment:
This is tricky as this function is used in a bunch of different ways.  it's currently being used for thresholding masks at a certain percentile and is intended to be one sided.  I don't think we should change the defaults as we will have to fix every function that calls it.  I can see a case to be made for having a threshold function for statistics, but we already have one (nltools.stats.threshold).

Maybe add something like a 'side=upper' flag with options for ['upper','lower','both'].

I don't think it is a good idea to have a function default change based on the content of the data in this context.  It will likely lead to unintended behavior.

Reply:
Fixed as per what we discussed the other day, along with test added